pluggy.py

```diff
@@ -605,10 +623,10 @@ def varnames(func):
 
     assert "self" not in args  # best naming practises check?
     try:
-        cache["_varnames"] = args
+        cache["_varnames"] = args, defaults
     except TypeError:
```

Comment:
@nicoddemus yeah I never really understood the weird "storing arg names in the function's `dict`" in the first place. Can't we just have a single global cache which hashes by object id?

Maybe @hpk42 can comment?